src/OrbitGl/TrackHeader.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TrackHeader.h"
+
+#include "AccessibleCaptureViewElement.h"
+#include "GlCanvas.h"
+#include "Track.h"
+#include "TrackRenderHelper.h"
+
+namespace orbit_gl {
+
+TrackHeader::TrackHeader(CaptureViewElement* parent, const Viewport* viewport,
+                         const TimeGraphLayout* layout)
+    : CaptureViewElement(parent, viewport, layout) {
+  ORBIT_CHECK(dynamic_cast<Track*>(parent) != nullptr);
```

Comment:
I think we should make this constraint on a type level. Make parent a `Track*`. We can still pass it to the `CaptureViewElement`'s constructor. For type safety we could think of having the parent also stored as `Track`. `GetParent` in `CaptureViewElement` is already virtual. We might want to override it such that it returns `Track*`.

Reply:
Oh right, that makes a lot of sense and gets rid of most of the dynamic_cast calls... I didn't realize `GetParent()` was virtual, thanks.